queries/janet_simple/highlights.scm

```diff
@@ -0,0 +1,302 @@
+;; >> Explanation
+;; Parsers for lisps are a bit weird in that they just return the raw forms.
+;; This means we have to do a bit of extra work in the queries to get things
+;; highlighted as they should be.
+;;
+;; For the most part this means that some things have to be assigned multiple
+;; groups.
+;; By doing this we can add a basic capture and then later refine it with more
+;; specialized captures.
+;; This can mean that sometimes things are highlighted weirdly because they
+;; have multiple highlight groups applied to them.
+
+
+;; >> Literals
+
+(kwd_lit) @symbol
+(str_lit) @string @spell
+(long_str_lit) @string @spell
+(buf_lit) @string @spell
+(long_buf_lit) @string @spell
+(num_lit) @number
+(bool_lit) @boolean
+(nil_lit) @constant.builtin
+(comment) @comment @spell
+
+["'" "~"] @string.escape
+
+["|" "," ";"] @punctuation.special
+
+["{" "@{" "}"
+ "[" "@[" "]"
+ "(" "@(" ")"] @punctuation.bracket
+
+;; >> Symbols
+
+; General symbol highlighting
+(sym_lit) @variable
+
+; General function calls
+(par_tup_lit
+ .
+ (sym_lit) @function.call)
+
+(short_fn_lit
+ .
+ (sym_lit) @function.call)
+
+; Quoted symbols
+(quote_lit
+ (sym_lit) @symbol)
+
+(qq_lit
+ (sym_lit) @symbol)
+
+; Dynamic variables
+((sym_lit) @variable.builtin
+ (#lua-match? @variable.builtin "^[*].+[*]$"))
+
+; Operators
+((sym_lit) @operator
+ (#any-of? @operator
+  "%" "*" "+" "-" "/"
+  "<" "<=" "=" "==" ">" ">="))
+
+((sym_lit) @keyword.operator
+ (#any-of? @keyword.operator
+  "not" "not=" "and" "or"))
+
+; Definition
+((sym_lit) @keyword
+ (#any-of? @keyword
+  "def" "def-" "defdyn" "defglobal" "defmacro" "defmacro-"
+  "var" "var-" "varglobal"))
+
+((sym_lit) @keyword.function
+ (#match? @keyword.function "^(defn|defn-|fn|varfn)$"))
+
+; Comment
+((sym_lit) @comment
+ (#any-of? @comment "comment"))
+
+; Conditionals
+((sym_lit) @conditional
+ (#any-of? @conditional
+  "case" "cond"))
+
+((sym_lit) @conditional
+ (#match? @conditional "^if(\\-.*)?$"))
+
+((sym_lit) @conditional
+ (#match? @conditional "^when(\\-.*)?$"))
+
+; Repeats
+((sym_lit) @repeat
+ (#any-of? @repeat
+  "for" "forever" "forv" "loop" "repeat" "while"))
+
+; Exception
+((sym_lit) @exception
+ (#any-of? @exception "error" "errorf" "try"))
+
+; Includes
+((sym_lit) @include
+ (#any-of? @include "import" "require" "use"))
+
+; Builtin macros
+;; TODO: Do all these items belong here?
```

Comment:
I  don't think they do. :)

Reply:
Ah, I think I failed to delete that comment from the original file :)

IIUC, it was added in this commit: https://github.com/nvim-treesitter/nvim-treesitter/commit/1099809d706c2661440386330a13efbb66a8e927

I think all of the listed things are macros -- IIRC I took [the list I'm using in janet-ts-mode for macros](https://github.com/sogaiu/janet-ts-mode/blob/140fdb3f8ab2c1429b444f26f03e047f7077b619/janet-ts-mode.el#L189-L223) and duplicated it.

In janet-ts-mode I wanted to make a distinction among special forms, functions, and macros.  I have found this to be helpful when working with lisps -- especially when learning a new one or switching among multiple.

There is overlap in the macros and functions portions with other bits in the `highligthts.scm` file such as: https://github.com/nvim-treesitter/nvim-treesitter/pull/4695/commits/defedcd66ddec761bdb895325cc7eec63d302bef#diff-020f6c8ceab2f6eb6dae784ba05fd3a4b3293ac6b4593abe35470dace40bd5dbR59-R67

I retained those and some other parts because I think they were added to Clojure's highlights.scm -- although I was the original author of that file, its current form has had much work done on it by other folks.  I don't really understand if there are implications for things being identified as `@operator` or `@keyword` (apart from highlighting) so I thought I'd leave these bits in.  For comparison in the Emacs tree-sitter mode, they are not specifically identified as such (or something similar) so I don't think the distinctions are used anywhere.

Perhaps I should remove them and if people want them they can be added back?